src/fft_common.f90

```diff
@@ -2,39 +2,45 @@
 
 ! This file contains common code shared by all FFT engines
 
-integer, save :: format                 ! input X-pencil or Z-pencil
-
-! The libary can only be initialised once
-logical, save :: initialised = .false.
+integer, pointer, save :: format => null()  ! input X-pencil or Z-pencil
 
 ! Global size of the FFT
-integer, save :: nx_fft, ny_fft, nz_fft
+integer, pointer, save :: nx_fft => null(), ny_fft => null(), nz_fft => null()
 
 ! 2D processor grid
 ! FIXME this is already available in the module decomp_2d
 integer, save, dimension(2) :: dims
 
 ! Decomposition objects
 TYPE(DECOMP_INFO), pointer, save :: ph => null()  ! physical space
-TYPE(DECOMP_INFO), target, save :: sp  ! spectral space
+TYPE(DECOMP_INFO), pointer, save :: sp => null()  ! spectral space
```

Comment:
was the `target` attribute wrong before?

Reply:
No